server/service/src/report/report_service.rs

```diff
@@ -325,16 +324,88 @@ fn get_report(ctx: &ServiceContext, id: &str) -> Result<Report, RepositoryError>
 
 fn query_reports(
     ctx: &ServiceContext,
-    pagination: Option<PaginationOption>,
     filter: Option<ReportFilter>,
     sort: Option<ReportSort>,
 ) -> Result<Vec<Report>, ListError> {
+    let app_version: Version = Version::from_package_json();
+
     let repo = ReportRepository::new(&ctx.connection);
-    let pagination = get_default_pagination(pagination, MAX_LIMIT, MIN_LIMIT)?;
     let filter = filter
         .unwrap_or_default()
         .r#type(ReportType::OmSupply.equal_to());
-    Ok(repo.query(pagination, Some(filter.clone()), sort)?)
+
+    let reports_to_show_meta_data = report_filter_method(
+        repo.query_meta_data(Some(filter.clone()), None)?,
+        app_version,
+    );
```

Comment:
Added querying all for metadata first with a sub selection of code, version, id, and is_custom. I thought this would help if our report list gets long and we aren't wanting to query all reports and filter out later.

Reply:
> Added querying all for metadata first with a sub selection of code, version, id, and is_custom. I thought this would help if our report list gets long and we aren't wanting to query all reports and filter out later.

That's awesome, thank you